common/component/kafka/auth.go

```diff
@@ -111,16 +113,21 @@ func updateAWSIAMAuthInfo(ctx context.Context, config *sarama.Config, metadata *
 }
 
 type mskAccessTokenProvider struct {
-	ctx          context.Context
-	accessKey    string
-	secretKey    string
-	sessionToken string
-	region       string
+	ctx                  context.Context
+	generateTokenTimeout time.Duration
+	accessKey            string
+	secretKey            string
+	sessionToken         string
+	region               string
 }
 
+// this function can't use the context passed on Init because that context would be cancelled right after Init
 func (m *mskAccessTokenProvider) Token() (*sarama.AccessToken, error) {
```

Comment:
```suggestion
func (m *mskAccessTokenProvider) Token() (*sarama.AccessToken, error) {
    // this function can't use the context passed on Init because that context would be cancelled right after Init
```